src/databasetasks.cpp

```diff
@@ -71,11 +71,11 @@ void DatabaseTasks::runTask(const DatabaseTask& task)
 	bool success;
 	DBResult_ptr result;
 	if (task.store) {
-		result = db.storeQuery(task.query);
+		result = Database::storeQuery(task.query);
```

Comment:
An option is having a `thread_local` mysql handle, so each thread that wants to perform queries sets up it's own connection.